mmcif/io/BinaryCifWriter.py

```diff
@@ -148,9 +153,30 @@ def __getAttributeType(self, dObj, atName):
                 logger.error("Undefined type for category %s attribute %s - Will treat as string", dObj.getName(), atName)
         else:
             dataType = "integer" if "int" in cifDataType else "float" if cifPrimitiveType == "numb" else "string"
+
+        # Only if applying types, do we allow Mol* hints
+        if self.__applyTypes and self.__applyMolStarTypes:
+            dch = DataCategoryHints()
```

Comment:
Would it be more efficient to set this up in `__init__` so that it doesn't have to re-instantiate every time? (e.g., set it as a `self.__dch` up above?)

Reply:
Sure -- likely we will need to do this.  Let me correct.